### PR TITLE
Fixing cache accessor

### PR DESCRIPTION
**Pull request by abhidnya13**
#135  was opened to fix the usage of `noopCacheAccessor`. The PR breaks the integration tests due to a null pointer exception for confidential client with client secret scenario. 
This PR is opened to fix the bug in `base.go` instead of making the same changes in both `public.go` and `confidential.go`

### CODE REVIEW

File: apps/internal/base/base.go [hunk 1]
```diff
@@ -123,11 +123,13 @@ func New(clientID string, authorityURI string, cacheAccessor cache.ExportReplace
 		return Client{}, err
 	}
 	authParams := authority.NewAuthParams(clientID, authInfo)
-
+	if cacheAccessor == nil {
```
Comment (keegan-caruso) [inline apps/internal/base/base.go:126]: might be worth documenting this in the comment for the method

**Review by keegan-caruso — APPROVED** [review]

File: apps/internal/base/base.go (hunk 1 above)
Comment (element-of-surprise) [reply apps/internal/base/base.go:126]: The other option is to make this a Go style option:

```go
func Option func(c *Client)

type WithCacheAccessor(ca cache.ExportReplace) Option {
  return func(c *Client) {
    if c == nil {
      return
    }
    c.cacheAccessor = ca
}

func New(clientID, authorityURI, token *oauth.Client, options ...Option) (Client, error) {
  ...
  client := Client {
    ...
   cacheAccessor: noopCacheAccessor{}
  }
  for _, o := range options {
    o(&client)
  }
  return client, nil
```

Then its a true option (which I prefer over accepting nils).

Comment (abhidnya13) [reply apps/internal/base/base.go]: @element-of-surprise , Tried changing this to accept it as an option in the latest commit.  
What do you think ?

Comment (keegan-caruso) [reply apps/internal/base/base.go]: Looks better, thanks

Comment (abhidnya13) [reply apps/internal/base/base.go]: Merging this in.

**Pull request merged**